File: frontend/src/services/DestinationService.js
import axios from "../custom-axios/axiosInstance";

const DestinationService={
    addNewDestination:(city,country,destinationImage,destinationThumbnail)=>{
        return axios.post('/api/destinations/addNewDestination',{
            "city":city,
            "country":country,
            "destinationImage":destinationImage,
            "destinationThumbnail":destinationThumbnail
        })
    },

    getAllDestinations:()=>{
        return axios.get('/api/destinations/getAllDestinations')
    },

    getDestinationDetails:(destinationId)=>{
        return axios.get(`/api/destinations/getDestinationDetails/${destinationId}`)
    },

    getWeatherDataForDestination:(destinationId)=>{
        return axios.get(`/api/destinations/getWeatherConditions/${destinationId}`)
    },

    getDestinationDetailsFromDbpedia:(city,country)=>{
        return axios.get(`/api/destinations/getDestinationDetailsFromDbpedia/${city}/${country}`)
    },

    getMuseumsForDestination:(destination)=>{
        return axios.get(`/api/destinations/getMuseumsForDestination/${destination}`)
    },

    getAttractionsForDestination:(destination)=>{
        return axios.get(`/api/destinations/getAttractionsForDestination/${destination}`)
    },

    getRestaurantsForDestination:(destination)=>{
        return axios.get(`/api/destinations/getRestaurantsForDestination/${destination}`)
    },

    getBarsForDestination:(destination)=>{
        return axios.get(`/api/destinations/getBarsForDestination/${destination}`)
    },

    getNightClubsForDestination:(destination)=>{
        return axios.get(`/api/destinations/getNightClubsForDestination/${destination}`)
    },

    getTopThreeDestinations:()=>{
        return axios.get(`/api/destinations/getTopThreeDestinations`)
    }
}

export default DestinationService;

File: backend/src/main/java/com/example/backend/service/DestinationService.java
package com.example.backend.service;

import com.example.backend.model.Destination;
import com.example.backend.model.dto.TouristAttractionDto;
import com.example.backend.model.dto.DestinationDto;
import com.example.backend.model.forms.DestinationForm;

import java.util.List;
import java.util.Optional;

public interface DestinationService {
    Optional<Destination> addNewDestination(DestinationForm destinationForm);
    List<Destination> getAllDestinations();
    List<Destination> findTopThreeDestinations();
    Optional<Destination> getDestination(String destinationId);
    DestinationDto getDestinationDetailsFromDbpedia(String city, String country);
    List<TouristAttractionDto> getMuseumsForDestination(String city);
    List<TouristAttractionDto> getAttractionsForDestination(String city);
    List<TouristAttractionDto> getRestaurantsForDestination(String city);
    List<TouristAttractionDto> getBarsForDestination(String city);
    List<TouristAttractionDto> getNightClubsForDestination(String city);

}
